src/pages/RouteView.tsx

```diff
@@ -0,0 +1,141 @@
+import { Box, Button, Center, Flex, Text, VStack } from 'native-base';
+import { NavBar } from '../components/NavigationBar';
+import { createSearchParams, useNavigate, useSearchParams } from 'react-router-dom';
+import { getCurrentUser, getRouteById } from '../xplat/api';
+import { useQuery } from 'react-query';
+import { NaturalRules, Route, RouteStatus, User, UserStatus, invalidateDocRefId } from '../xplat/types';
+import { queryClient } from '../App';
+import placeholder_image from '../placeholder_image.jpg';
+import { useEffect, useState } from 'react';
+import { buildUserByIDFetcher } from '../utils/queries';
+
+const RouteView = () => {
+  const [params] = useSearchParams();
+  const navigate = useNavigate();
+  const [hasUserUID, setHasUserUID] = useState<boolean>(false);
+  const [userUID, setUserUID] = useState<string>('');
+
+  const hasRouteUID: boolean = params.has('uid');
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const routeUID: string = (hasRouteUID ? params.get('uid')! : '');
+
+  const routeQuery = useQuery(
+    ['routes', routeUID],
+    Route.buildFetcherFromDocRefId(routeUID),
+    {
+      enabled: hasRouteUID,
+    }
+  );
+
+  useEffect(() => {
+    const fetchUserUID = async () => {
```

Comment:
I saw that you merged the AuthContext PR into this, so you should be using that instead of calling getCurrentUser() since that is an extra read we don't need.